cpp/cmake/3rdparty.cmake

```diff
@@ -0,0 +1,94 @@
+find_package(Git QUIET)
+
+if(NOT ${CODING_CONV_PREFIX}_3RDPARTY_DIR)
+  set(${CODING_CONV_PREFIX}_3RDPARTY_DIR 3rdparty)
+  set(${CODING_CONV_PREFIX}_3RDPARTY_DIR
+      3rdparty
+      PARENT_SCOPE)
+endif()
+
+# initialize submodule with given path
+function(bbp_init_git_submodule path)
+  if(NOT ${GIT_FOUND})
+    message(
+      FATAL_ERROR "git not found and ${path} submodule not cloned (use git clone --recursive)")
+  endif()
+  message(
+    STATUS "Fetching git submodule ${path}: running git submodule update --init --recursive ${path}"
+  )
+  execute_process(
+    COMMAND
+      ${GIT_EXECUTABLE} submodule update --init --recursive -- ${path}
+    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+    RESULT_VARIABLE git_submodule_status)
+  if(NOT git_submodule_status EQUAL 0)
+    message(FATAL_ERROR "Could not clone git submodule ${path}")
+  endif()
+endfunction()
+
+# use a git submodule
+#
+# bbp_git_submodule(source_dir
+#                   [DISABLED]
```

Comment:
so here i would also provide `[RECURSIVE <on|off>]` or maybe `[NORECURSIVE]`, so it's recursive when not set.